application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java

```diff
@@ -54,19 +67,41 @@ static Button createDeleteButton(String componentId) {
      *
      * @param id the id of the tag to check
      * @param event the event to send messages with
+     * @param componentIdGenerator the component id generator to generate the button ids with
+     * @param userToReplyTo the user that was originally meant to be replied to when the tag command
+     *        was invoked
      * @return whether the given tag is unknown to the system
      */
-    boolean handleIsUnknownTag(String id, IReplyCallback event) {
+    boolean handleIsUnknownTag(String id, IReplyCallback event,
+            ComponentIdGenerator componentIdGenerator, @Nullable OptionMapping userToReplyTo) {
         if (hasTag(id)) {
             return false;
         }
-        String suggestionText = StringDistances.closestMatch(id, getAllIds())
-            .map(", did you perhaps mean '%s'?"::formatted)
-            .orElse(".");
 
-        event.reply("Could not find any tag with id '%s'%s".formatted(id, suggestionText))
-            .setEphemeral(true)
-            .queue();
+        List<String> candidates = getAllIds().stream()
+            .sorted(Comparator
+                .comparingInt(candidate -> StringDistances.editDistance(id, candidate)))
+            .limit(TAG_SUGGESTIONS_AMOUNT)
+            .toList();
```

Comment:
Id suggest using a PQ with k-sort instead. Would be faster if the amount of tags increases further.

```java
Queue<String> closestMatches = new PriorityQueue<>(Comparator...);
closestMatches.addAll(getAllIds());

List<String> suggestions = Stream.generate(closestMatches::poll)
  .limit(TAG_SUGGESTIONS_AMOUNT)
  .toList();
```